crates/javy/src/config.rs

```diff
@@ -161,10 +170,62 @@ impl Config {
         self
     }
 
+    /// Whether the `Javy.JSON` intrinsic will be available.
+    /// Disabled by default.
+    /// This setting requires the `json` crate feature to be enabled.
+    #[cfg(feature = "json")]
+    pub fn javy_json(&mut self, enable: bool) -> &mut Self {
+        self.javy_intrinsics.set(JavyIntrinsics::JSON, enable);
+        self
+    }
+
     /// Enables whether the output of console.log will be redirected to
     /// `stderr`.
     pub fn redirect_stdout_to_stderr(&mut self, enable: bool) -> &mut Self {
         self.redirect_stdout_to_stderr = enable;
         self
     }
+
+    /// Whether to override the implemenation of JSON.parse and JSON.stringify
+    /// with a Rust implementation that uses a combination of Serde transcoding
+    /// serde_json and simd_json for improved performance.
+    /// This setting requires the `JSON` intrinsic to be enabled and the `json`
+    /// crate feature to be enabled as well.
+    /// Disabled by default.
+    #[cfg(feature = "json")]
+    pub fn override_json_parse_and_stringify(&mut self, enable: bool) -> &mut Self {
+        self.override_json_parse_and_stringify = enable;
+        self
+    }
+
+    pub(crate) fn validate(self) -> Result<Self> {
```

Comment:
Nit: Is there a reason this can't take a reference to `self` and return that same reference?

Reply:
Usually these validate methods take ownership of self to force the caller to coerce `&self` or `&mut self` to `self` and avoid any unwanted usage or mutation of the config after validation. That's the reason why I opted to take in `self`